metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertCommentCodeAction.scala

```diff
@@ -0,0 +1,138 @@
+            if (cursorIsInTheSingleLineComment) {
+              Trees.defaultTokenizerDialect(content).tokenize.toOption
+            } else None
+          }
+      else None
+
+    tokenized
+      .flatMap { tokens =>
+        computeChanges(tokens, range)
+          .map { changes =>
+            val action = CodeActionBuilder.build(
+              title = ConvertCommentCodeAction.Title,
+              kind = this.kind,
+              changes = List(path -> changes),
+            )
+            List(action)
+          }
+      }
+      .getOrElse(List.empty)
+  }
+
+  private def computeChanges(
+      tokens: Tokens,
+      range: l.Range,
+  ): Option[List[l.TextEdit]] = {
+    val where = tokens.lastIndexWhere(t =>
+      t.pos.startLine == range.getStart.getLine
+        && t.pos.endLine == range.getEnd.getLine
+        && t.pos.startColumn < range.getStart.getCharacter
+    )
+    val commentBeforeCursor = collectContinuousComments(
+      tokens
+        .take(where)
+        .reverse
+    ).reverse
+
+    val commentAfterCursor = collectContinuousComments(tokens.drop(where))
+    val commentStart = commentBeforeCursor.headOption
+      .map(_.pos.toLsp.getStart)
+      .getOrElse(range.getStart)
+    val commentEnd = commentAfterCursor.lastOption
+      .map(_.pos.toLsp.getEnd())
+      .getOrElse(range.getEnd())
+
+    val commentTokens = commentBeforeCursor ++ commentAfterCursor
+    if (commentTokens.nonEmpty) {
+      val replaceText =
+        commentTokens
+          .map(_.value)
+          .mkString("/*", "\n *", " */")
+      commentStart.setCharacter(tokens(where).pos.startColumn)
+      val pos = new l.Range(commentStart, commentEnd)
+      Some(List(new l.TextEdit(pos, replaceText)))
+    } else {
+      None
+    }
+  }
+  private def collectContinuousComments(tokens: Seq[Token]) = {
+    sealed trait State {
+      def fold[T](isEmpty: => T)(nonEmpty: Seq[Token.Comment] => T): T = {
+        this match {
+          case State.Empty => isEmpty
+          case State.Continue(acc, _) => nonEmpty(acc)
+          case State.Stop(acc) => nonEmpty(acc)
+        }
+      }
+    }
+    object State {
+      case object Empty extends State
+      case class Continue(acc: Seq[Token.Comment], last: Token) extends State
+      case class Stop(acc: Seq[Token.Comment]) extends State
+    }
+    tokens
+      .foldLeft(State.Empty: State) {
+        case (s: State.Stop, _) => s
+        case (State.Empty, item) =>
+          item match {
+            case newComment: Token.Comment =>
+              State.Continue(Seq(newComment), newComment)
+            case _: Token.Whitespace => State.Empty
+            case _: Token.BOF => State.Empty
+            case _ => State.Stop(Seq.empty)
+          }
+        case (State.Continue(acc, last), item) =>
+          // comments can be separated by EOL tokens and other whitespace characters
+          (last, item) match {
+            case (_: Token.EOL, newComment: Token.Comment) =>
```

Comment:
You never really check if those are single line comments. How would that work for:
```
// <<>>comment
/* other comment */
```
or
```
/* smth about file://SomeFile <<>>and smth */
```

Reply:
True, I also realized that post-factum but it seemed that it will work. 
The first one will merge comments together. 

But the second one is indeed trickier, it will actually contribute action but the action won't do anything. I need to think about it.

Reply:
This is now fixed.